src/main/java/com/patchworkmc/mapping/IntermediaryHolder.java

```diff
@@ -0,0 +1,71 @@
+package com.patchworkmc.mapping;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import net.fabricmc.tinyremapper.IMappingProvider;
+
+/**
+ * A lazily loading representation of all Minecraft names and descriptors.
+ */
+public class IntermediaryHolder {
+	private final HashMap<String, Map<String, Member>> mappings = new HashMap<>();
+
+	private final IMappingProvider inverted;
+
+	private boolean loaded = false;
+
+	public IntermediaryHolder(IMappingProvider inverted) {
+		this.inverted = inverted;
+	}
+
+	public Map<String, Member> getMappings(String owner) {
+		// We load lazily here so that if this class isn't needed we can save some time and memory
+		if (!this.loaded) {
+			inverted.load(new Acceptor());
+			loaded = true;
+		}
+
+		return mappings.get(owner);
+	}
+
+	// kotlin-style object to hold fields.
```

Comment:
a better way to refer to this is `POJO` (plain old java object)
```suggestion
	// kotlin-style object to hold fields and methods.
```